File: 03_Binary Representation of a Number/Ideal Solutions/binaryRepresentation.java
import java.io.*;
import java.lang.*;
import java.util.*;


class Main{
	//This function changes the array into the string.
	public static String arrayToString(int[] arr){
		StringBuffer ans = new StringBuffer();
		for(int i=0;i<arr.length;i++)ans.append(arr[i]);
		return ans.toString();
	}
	//This function converts a number into its binary representation.
	public static String binaryRepresentation(int n){
		int[] arr = new int[32]; 
		Arrays.fill(arr,0); //Filling the entire arrays with a default value of 0.
		int len = arr.length-1;
		while(n > 0){
			int temp = n % 2;
			arr[len] = temp;
          	len--;
			n = n/2;
		}
		String ans = arrayToString(arr);
		return ans;

	}
	//Main Function
	public static void main(String[] args){
		int n = 7;
		String ans = binaryRepresentation(n);
      	System.out.println(ans);	
	}

}

File: 03_Binary Representation of a Number/Ideal Solutions/binaryRepresentation.cpp
#include<bits/stdc++.h>
using namespace std;
void binaryRepresentation(int N) {
	// Function to print the binary representation of a 32-bit integer

	// For a 32 bit integer bit positions are 0-31 bits
	for (int i = 31; i >= 0; i--) {
		if (N & (1 << i))
			cout << 1;
		else
			cout << 0;
	}
	cout << endl;
}
int main() {
	int N;
	N = 7;
	binaryRepresentation(N);
	return 0;
}
